src/styles/foundation/utilities.scss

```diff
@@ -1,27 +1,31 @@
-$default-browser-font-size: 16px;
-$base-font-size: 10px;
+$root-font-size: 16px;
 
-/// Returns the value in rem for a given pixel value.
-/// @param {Number} $value - The pixel value to be converted.
+/// Returns the value in rem for a given length value.
+/// Note: converting ems only works for elements that have had no font-size changes.
+/// @param {Number} $value - The value (em, rem, or px) to be converted.
 /// @return {Number} The converted value in rem.
 
 @function rem($value) {
   $unit: unit($value);
+  $current-font-size: $root-font-size;
 
   @if $value == 0 {
     @return 0;
   } @else if $unit == 'rem' {
     @return $value;
   } @else if $unit == 'px' {
-    @return $value / $base-font-size * 1rem;
+    // e.g. 16px × 1rem/16px = 1rem
+    @return $value * (1rem / $root-font-size);
   } @else if $unit == 'em' {
-    @return $unit / 1em * 1rem;
+    // This is a fudge; it assumes 1em == 1rem
+    @return $value / 1em * 1rem;
```

Comment:
Fixing this bug means we'll now have to support it.

Since we know nobody is using this (or they'd have submitted an issue), I vote for removing this part completely:

```diff
- } @else if $unit == 'em' {
-    @return $unit / 1em * 1rem;

# and also:
-   @error 'Value must be in px, em, or rem.';
+   @error 'Value must be in px, or rem.';
```

Reply:
The thing is we’re supporting this API already, and fixing it is low cost.

I’d be all for removing em unit conversion if it wasn’t needed. However, [we convert _to_ ems in our own code](https://github.com/Shopify/polaris-react/blob/6b224cdcfe95f2b45adc0f282e3ddb7757e71b8c/src/styles/shared/breakpoints.scss#L5), so we have to support that. The most we could do is remove converting _from_ em units. That seems like a strangely asymmetric API to me.

Is there really any downside to just shipping this bug fix?

Reply:
Thanks Kaelig. I see where you’re coming from, and I appreciate it. In general I definitely support not adding (and deleting!) code we don’t use.